src/components/CldImage/CldImage.vue

```diff
@@ -107,19 +96,61 @@ export default {
       default: ""
     }
   },
+  provide() {
+    return {
+      registerTransformation: this.registerTransformation,
+      getTransformOptions: () => {
+        return this.transformOptions;
+      },
+      configuration: this.configuration,
+      publicId: this.publicId,
+      isImageLoaded: () => {
+        return this.imageLoaded;
+      },
+      registerPlaceHolder: () => {
+        this.hasPlaceholderComponent = true;
+      },
+      getImageWidth: () => {
+        // This gets populated from the <cld-image width=500/> tag
```

Comment:
This can't come from other places? like child cld-transformation?

Reply:
This is identical to the angular implementation

I think we had a problem as cld-transfomration can in theory, have multiple "widths" set, (you can chain them) 
this makes it quite difficult to know what 'width' we want.